PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebLauncher.kt

```diff
@@ -0,0 +1,162 @@
+package com.paypal.android.paypalwebpayments
+
+import android.net.Uri
+import androidx.fragment.app.FragmentActivity
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.BrowserSwitchStatus
+import com.paypal.android.corepayments.APIClientError
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.Environment
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.DEEP_LINK_PARAM_APPROVAL_SESSION_ID
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.DEEP_LINK_PARAM_APPROVAL_TOKEN_ID
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.VAULT_HOST
+import com.paypal.android.paypalwebpayments.errors.PayPalWebCheckoutError
+import org.json.JSONObject
+
+internal class PayPalWebLauncher(
```

Comment:
This extracts browser switching behavior from `PayPalWebCheckoutClient` and combines it with `BrowserSwitchHelper` logic to encapsulate all browser switching logic in one place. In `v2` we'll be able to make this class public and adjust the API so that it works in tandem with `PayPalWebCheckoutClient`.